From b6f07f34b6275b08164740f30ec12bce6665afb4 Mon Sep 17 00:00:00 2001
From: Amit Mittal <amit.mittal@st.com>
Date: Tue, 30 Jul 2024 11:58:32 +0530
Subject: [PATCH] dt-bindings: remoteproc: Add pm domain, property
 keep-power-in-suspend

Add the PM domains cluster and retention.

Change-Id: If03e7b22b9d23ec5e3de036e9384ff44682f11dc
Signed-off-by: Amit Mittal <amit.mittal@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/396286
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 .../bindings/remoteproc/st,stm32-rproc.yaml       | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/Documentation/devicetree/bindings/remoteproc/st,stm32-rproc.yaml b/Documentation/devicetree/bindings/remoteproc/st,stm32-rproc.yaml
index 79e623d24cae..87b598b398dd 100644
--- a/Documentation/devicetree/bindings/remoteproc/st,stm32-rproc.yaml
+++ b/Documentation/devicetree/bindings/remoteproc/st,stm32-rproc.yaml
@@ -147,6 +147,21 @@ properties:
       If defined, when remoteproc is probed, it loads the default firmware and
       starts the remote processor.
 
+  power-domains:
+    items:
+      - description: CLUSTER power domain
+      - description: RETENTION power domain
+
+  power-domain-names:
+    items:
+      - const: default
+      - const: sleep
+
+  keep-power-in-suspend:
+    $ref: /schemas/types.yaml#/definitions/flag
+    description:
+      SRAM only. Preserves power during a suspend/resume cycle.
+
 required:
   - compatible
   - reg
-- 
2.39.5

